### ISSUE TITLE
Type Mismatch in Returned Data Using generateServerClientUsingCookies Method from @aws-amplify/adapter-nextjs/data in Latest Version

### ISSUE THREAD

**Issue by gentamura** [aws-amplify/amplify-category-api#2643]
### Environment information

```plain text
System:
  OS: macOS 14.5
  CPU: (8) arm64 Apple M2
  Memory: 187.23 MB / 24.00 GB
  Shell: /bin/zsh
Binaries:
  Node: 20.10.0 - ~/.anyenv/envs/NODENV/versions/20.10.0/bin/node
  Yarn: 1.22.21 - ~/.anyenv/envs/NODENV/versions/20.10.0/bin/yarn
  npm: 10.2.3 - ~/.anyenv/envs/NODENV/versions/20.10.0/bin/npm
  pnpm: 9.3.0 - ~/Library/pnpm/pnpm
NPM Packages:
  @aws-amplify/backend: 1.0.3
  @aws-amplify/backend-cli: 1.0.4
  aws-amplify: 6.3.7
  aws-cdk: Not Found
  aws-cdk-lib: Not Found
  typescript: 5.4.5
AWS environment variables:
  AWS_STS_REGIONAL_ENDPOINTS = regional
  AWS_NODEJS_CONNECTION_REUSE_ENABLED = 1
  AWS_SDK_LOAD_CONFIG = 1
No CDK environment variables
```

### Data packages

```plain text
aws-amplify-gen2@0.1.0 /Users/gentamura/workspace/playground/amplify-next-template
├─┬ @aws-amplify/backend-cli@1.0.4
│ └─┬ @aws-amplify/schema-generator@1.0.0
│   └── @aws-amplify/graphql-schema-generator@0.8.6
└─┬ @aws-amplify/backend@1.0.3
  └─┬ @aws-amplify/backend-data@1.0.2
    └── @aws-amplify/data-construct@1.8.5
```

### Description

When I used a `create` method from `generateServerClientUsingCookies` imported from `@aws-amplify/adapter-nextjs/data` with this latest version, the data returned is the actual value, but the type is returned as `type ClientModel`. The type itself can be obtained with `Schema[’Todo']['type']`, but the type of the returned data is likely to be a metadata-only completion and not the type assumed.
I have created a reference repository for this issue, which you are welcome to use.
https://github.com/gentamura/amplify-next-template

The reference images are as follows.

![スクリーンショット 2024-06-20 21 37 06](https://github.com/aws-amplify/amplify-category-api/assets/1329464/487a67eb-bb92-4327-822d-214dd125ac5d)

![スクリーンショット 2024-06-20 21 37 34](https://github.com/aws-amplify/amplify-category-api/assets/1329464/82170eb0-f80a-45bc-81c0-000b69a2620a)

I remember that in previous versions the type was set correctly, so it would be appreciated if you could investigate this.

Thank you.

**Comment by chrisbonifacio** [comment]
Hi @gentamura thanks for raising this issue. We were able to reproduce it consistently and compared the types between the clients (normal, cookie based, and req/res based) and found that the types were not being returned correctly from the library. Please track https://github.com/aws-amplify/amplify-api-next/pull/253 for updates on the fix.

**Comment by svidgen** [comment]
Hey @gentamura, this should be fixed in `@aws-amplify/data-schema@1.3.3`. Please pull the latest and confirm it resolves the problem:

```
npm update @aws-amplify/data-schema
```

(You may also need to [restart TS server](https://stackoverflow.com/a/64456017/779572).)

**Comment by gentamura** [comment]
@chrisbonifacio @svidgen 
 I can confirm that it has been fixed 🙌 
 Thanks for the quick fix 🫶
